File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # for i in nums:
        #     # return the number if count is zero
        #     if nums.count(i) == 1:
        #         return i

        res = nums[0]
        for x in range (1, len(nums)):
            res ^= nums[x]
        return res

File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        // take res as the first number
        int res = nums[0];
        // for every element after 1st element i.e index 1 to last
        for(int i=1; i< nums.length; i++){
// xor of same element results in zero and xor of 0 and number is number
            res ^= nums[i];
// so if number repeats the res becomes zero and after zero xor results in unique element
        }
        return res;
    }
}